src/create_update.c

```diff
@@ -506,13 +514,14 @@ int main(int argc, char **argv)
 			rename_detection(newm);
 #endif
 			/* Step 6b: otherwise, write out the manifest */
-			old_deleted = remove_old_deleted_files(oldm, newm);
+			old_deleted = remove_deprecated_files(oldm, newm, comp_deleted);
+			old_ghosted = remove_deprecated_files(oldm, newm, comp_ghosted);
 			sort_manifest_by_version(newm);
 			type_change_detection(newm);
 			newfiles = prune_manifest(newm);
-			if (newfiles > 0 || old_deleted > 0 || changed_includes(oldm, newm)) {
-				LOG(NULL, "", "%s component has changes (%d new, %d deleted), writing out new manifest", group, newfiles, old_deleted);
-				printf("%s component has changes (%d new, %d deleted), writing out new manifest\n", group, newfiles, old_deleted);
+			if (newfiles > 0 || old_deleted > 0 || old_ghosted > 0 || changed_includes(oldm, newm)) {
+				LOG(NULL, "", "%s component has changes (%d new, %d deleted, %d ghosted), writing out new manifest", group, newfiles, old_deleted, old_ghosted);
```

Comment:
while i agree, it looks ugly. i'd rather file a separate issue, pointing out examples. because, firstly, i doubt changes in this PR affected all occurrences of duplicating output; secondly, it's our of the scope of this particular change to modify the behavior.